### PR TITLE
English idioms

### PR DESCRIPTION
**Pull request by HAKSOAT**
Add revscoring.languages.english.get_idioms method

Idioms can be problematic in formal articles. So,
a list of idioms are needed to track them when they appear
in the articles.

Task: [T205545](https://phabricator.wikimedia.org/T205545)

### CODE REVIEW

File: revscoring/languages/english.py [hunk 1]
```diff
@@ -264,3 +266,23 @@ def safe_dictionary_check(word):
...
+        list='categorymembers',
+        cmtitle="Category:English idioms",
+        formatversion=2,
+        continuation=True)
+
+    idioms = []
+    for doc in results:
+        for page_doc in doc['query']['categorymembers']:
+            idiom = page_doc['title']
+            if 'Category:English' not in idiom:
```
Comment (halfak) [inline revscoring/languages/english.py]: What's the meaning of this check?

Comment (HAKSOAT) [reply revscoring/languages/english.py]: In the last five or so results.... There are values which are not idioms that have Category:English as suffixes. That's why I checked to ensure they are not added to the list.

**Comment by halfak** [comment]
This is great.  Thanks for your work.  I think we'll want to turn this into an asset that we can load in the enwiki.py file rather than making an API request every time we load the file.  

Maybe you could make a utility in `revscoring/utilities` that prints out the list of idioms.  Then we can turn that into a file we load in enwiki.py.  What do you think?

**Comment by HAKSOAT** [comment]
I agree. I wasn't comfortable putting that function there in the first place.
 Just had to do something and get feedback. I'll look into your suggestion and get back to you.

File: revscoring/languages/english.py (hunk 1 above)
Comment (halfak) [reply revscoring/languages/english.py]: Aha!  That makes sense.  It might me work adding a comment explaining this.  Otherwise, it looks good.

**Comment by halfak** [comment]
Sounds great!  Thanks for your work :)

**Comment by HAKSOAT** [comment]
Hello @halfak How about the idioms are extracted and redirected into a text file, such that we have an idiom per line and can read that data into memory when needed.

There are 7943 idioms in the list.

**Comment by halfak** [comment]
That sounds great.

**Comment by HAKSOAT** [comment]
I think I just messed up the PR. I can see commits I didn't make.

**Comment by HAKSOAT** [comment]
Hello @halfak The idioms are in the revscoring/utilities folder. Stored as idioms.txt.

**Comment by halfak** [comment]
Yeah.  It looks like you might need a rebase.  I can dive into this to get it cleaned up, but I'm not sure I'll have time for that soon.  Alternatively, you could start a fresh PR and copy-paste your work.

**Comment by HAKSOAT** [comment]
Hello @halfak I have resolved the issue.

**Comment by halfak** [comment]
Nice work!  This looks great.  Could you try to adapt your script in "utilities" to look like the others.  Once you do, you can run your script from the base directory by running:
```
./utility fetch_idioms
```

You'll note that the other utilities write to sys.stdout by default.  I'd suggest copy-pasting that boilerplate.  Note that we're using `docopt` in the other utilities.  Docopt parses the script documentation at the top of the page in order to figure out how to process the arguments.

**Comment by HAKSOAT** [comment]
Okay. I'll work on that. Thanks.

File: revscoring/utilities/fetch_idioms.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
+"""
+``revscoring fetch_idioms -h``
+::
+
+    Gets a list of idioms.
+
+    Usage:
+        fetch_text [--output=<path>]
```
Comment (halfak) [inline revscoring/utilities/fetch_idioms.py]: fetch_idioms

File: revscoring/utilities/fetch_idioms.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+"""
+``revscoring fetch_idioms -h``
+::
+
+    Gets a list of idioms.
```
Comment (halfak) [inline revscoring/utilities/fetch_idioms.py]: Gets a list of *English language* idioms from en.wiktionary.org

File: revscoring/utilities/fetch_idioms.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    Usage:
+        fetch_text [--output=<path>]
+                   [--verbose] [--debug]
+
+    Options:
+        -h --help        Print this documentation
+        --output=<path>  Path to a file to write the idioms
+                         [default: revscoring/utilities/idioms.txt]
```
Comment (halfak) [inline revscoring/utilities/fetch_idioms.py]: Should default to <stdout> like the other utilities.

File: revscoring/utilities/fetch_idioms.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+def save(file_path, idioms):
+    with open(file_path, 'w') as file:
+        for idiom in idioms:
+            file.write(idiom + "\n")
+
+
+def run(output, verbose):
+    if verbose:
+        sys.stdout.write('Fetching idioms...')
```
Comment (halfak) [inline revscoring/utilities/fetch_idioms.py]: You probably want something like:
```
logger.info("Fetching idioms...")
```

File: revscoring/utilities/fetch_idioms.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    idioms = fetch()
+    if verbose:
+        sys.stdout.write('Saving idioms...')
```
Comment (halfak) [inline revscoring/utilities/fetch_idioms.py]: Same here. 
```
logger.info("Saving idioms...") 
```

**Comment by halfak** [comment]
Looks good.  I left a few notes.  I think we'll want to move the output to a new directory.  Maybe we can put it into the base directory and call it "asserts/".

**Comment by HAKSOAT** [comment]
@halfak You mean "assets" instead of "asserts" right?

**Comment by HAKSOAT** [comment]
I've made changes. I used "assets" instead of "asserts".

**Comment by halfak** [comment]
Yes!  Thank you.

**Comment by halfak** [comment]
Some notes from IRC: 

We need to first add the idioms to revscoring/languages/english.py

Check out what we do with badwords and informals for an example for how we can construct a set of "RegexMatches" features. 

You should be able to load in the idioms, construct the RegexMatches object and then we can use that to scan for idioms are use those counts as features. 

Once we get all of that worked out, we'll turn to the model repos like editquality, articlequality, etc.

**Comment by halfak** [comment]
I merged this manually.  not sure why github didn't notice.

**Pull request closed without merging**